File: Fences/FencesGame/Connection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FencesGame
{
    public class Connection
    {
        public Orientation Direction;
        public TileState Color;
        public int Row;
        public int Collumn;

        protected bool Equals(Connection other)
        {
            return Direction == other.Direction && Color == other.Color && Row == other.Row && Collumn == other.Collumn;
        }

        public override int GetHashCode()
        {
            unchecked
            {
                var hashCode = (int) Direction;
                hashCode = (hashCode*397) ^ (int) Color;
                hashCode = (hashCode*397) ^ Row;
                hashCode = (hashCode*397) ^ Collumn;
                return hashCode;
            }
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;

            return obj.GetType() == this.GetType() && Equals((Connection) obj);
        }
    }
}

File: FencesAndroid/app/src/main/java/com/example/fences_android/core/Connection.kt
package com.example.fences_android.core

class Connection {
    var Direction: Orientation = Orientation.Vertical;
    var Color: TileState = TileState.Empty;
    var Row: Int = 0;
    var Column: Int = 0;

    constructor(row: Int, col:Int, color: TileState, direction: Orientation) {
        Row = row;
        Column = col;
        Color = color;
        Direction = direction;
    }

    override fun equals(other: Any?): Boolean {
        if (other == null) {
            return false
        }

        val o = other as Connection
        return Row == o.Row && Column == o.Column && Color == o.Color && Direction == o.Direction
    }
}